Sources/CvUnit.cpp

```diff
@@ -7026,6 +7026,30 @@ bool CvUnit::canScrap() const
 }
 
 
+int CvUnit::calculateScrapValue() const
+{
+	float fCost = float(getUnitInfo().getProductionCost() * GC.getGameSpeedInfo(GC.getGame().getGameSpeedType()).getTrainPercent() / 100);
+	fCost += fCost * float(getUpgradeDiscount() / 100);
```

Comment:
The float cast here should be done on only the dividend or the divisor if you want a precision result from that division math. else it will be float(int / int) = float(int) which is pointless as the rounding down happens before you float cast it and because it will automatically make it a float as it is to be stored in a float variable there.

Reply:
Mmm, right. Will make this change and see if I made other odd choices of the same vein elsewhere, plus altering the pow() func. to how Bill suggested.